packages/ember-toucan-core/src/-private/assert-block-or-argument-exists.ts

```diff
@@ -0,0 +1,39 @@
+import { assert } from '@ember/debug';
+
+export type AssertBlockOrArg = {
+  blockExists: boolean;
+  argName: string;
+  arg?: string;
+  isRequired?: boolean;
```

Comment:
It may be helful to add some docs here for ourselves and future travelers:

```suggestion
  /**
   * The name of the argument that should be required.
   */
  arg?: string;
  /**
   * The name of the block or argument. Currently, the block and argument name must match.
   */
  argName: string;
  /**
   * Does the provided block exist?
   */
  blockExists: boolean;
  /**
   * Should this assertion be required, or is it optional?
   */
  isRequired?: boolean;
```

---

Another super nit - but `arg` and `argName` now that I'm looking at it _may_ be a little confusing on which is which.  I'm wondering if there's a different name _not_ including `arg` that could be used?  Would `name` work alone?  Something else?